routers/api/v1/utils/utils.go

```diff
@@ -0,0 +1,15 @@
+// Copyright 2017 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package utils
+
+import "code.gitea.io/gitea/modules/context"
+
+// UserID user ID of authenticated user, or 0 if not authenticated
+func UserID(ctx *context.APIContext) int64 {
```

Comment:
I would prefer the definition of a method like models.UserAccessLevel(ctx.User, repo) that check :
` if ctx.User == nil { ...` an return models.AccessLevel(